src/hyperion/external_interaction/callbacks/xray_centre/zocalo_callback.py

```diff
@@ -43,21 +41,27 @@ class XrayCentreZocaloCallback(CallbackBase):
 
     def __init__(
         self,
-        parameters: GridscanInternalParameters,
         ispyb_handler: GridscanISPyBCallback,
     ):
-        self.grid_position_to_motor_position: Callable[
-            [ndarray], ndarray
-        ] = parameters.experiment_params.grid_position_to_motor_position
         self.processing_start_time = 0.0
         self.processing_time = 0.0
         self.do_fgs_uid: Optional[str] = None
         self.ispyb: GridscanISPyBCallback = ispyb_handler
-        self.zocalo_interactor = ZocaloInteractor(
-            parameters.hyperion_params.zocalo_environment
-        )
 
     def start(self, doc: dict):
+        if doc.get("subplan_name") == GRIDSCAN_OUTER_PLAN:
+            LOGGER.info(
+                "Zocalo callback recieved start document with experiment parameters."
+            )
+            assert (
+                self.ispyb.params is not None
+            ), "ISPyB handler attached to Zocalo handler did not recieve parameters"
+            self.zocalo_interactor = ZocaloInteractor(
+                self.ispyb.params.hyperion_params.zocalo_environment
+            )
+            self.grid_position_to_motor_position: Callable[
+                [ndarray], ndarray
+            ] = self.ispyb.params.experiment_params.grid_position_to_motor_position
```

Comment:
Should: I don't like that we have to go through the ispyb object to get these, it seems really tightly coupled. Can we have a chat about a better way?

Reply:
I didn't think it was a problem since we're pulling ispyb IDs from there anyway, but we can just get it from the parameters, that's fine